Epic 3 - ReportApp/ReportApp/Services/Activity/ActivityReportGeneratorService.cs

```diff
@@ -0,0 +1,192 @@
+﻿using ClosedXML.Excel;
+using ClosedXML.Report;
+using ReportApp.Models.ActivityReport;
+using ReportApp.Models.Entity;
+
+namespace ReportApp.Services.Activity
+{
+    public class ActivityReportGeneratorService : IReportGenerator
+    {
+        private readonly ITemplateLoader _templateLoader;
+        private readonly IConfigurationLoader<ActivityReportConfiguration> _configurationLoader;
+        private readonly IReportSerializer<ActivityReportModel> _reportSerializer;
+
+        private Dictionary<string, Func<ActivityReportModel, object>> KeyValuePairs { get; set; }
+        private ActivityReportSettings Settings { get; set; }
+
+        public ActivityReportGeneratorService(
+            ITemplateLoader templateLoader,
+            IConfigurationLoader<ActivityReportConfiguration> configurationLoader,
+            IReportSerializer<ActivityReportModel> reportSerializer)
+        {
+            _templateLoader = templateLoader;
+            _configurationLoader = configurationLoader;
+            _reportSerializer = reportSerializer;
+            InitializeKeyValuePairs();
+        }
+
+        private void InitializeKeyValuePairs()
+        {
+            KeyValuePairs = new Dictionary<string, Func<ActivityReportModel, object>>
+            {
+                { "FirstName", r => r.ReportGeneratedFor.FirstName },
+                { "LastName", r => r.ReportGeneratedFor.LastName },
+                { "Day", r => r.WorkdayStartTime.ToShortDateString() },
+                { "Office", r => r.Office },
+                { "Additional Info", r => r.Complains }
+            };
+        }
+
+        public void GenerateReport(string jsonPath, string outputPath)
+        {
+            var template = _templateLoader.LoadTemplate("./Templates/ActivityReportTemplate.xlsx");
+            var configuration = _configurationLoader.LoadConfiguration("./ReportConfigurations/Activity.json");
+
+            var model = _reportSerializer.DeserializeReportModel(jsonPath);
+            FillSettings(model);
+            FillReportDataFromModel(template, configuration, model);
+            FillHeader(template, configuration);
+
+            template.Generate();
+            template.SaveAs(outputPath);
+        }
+
+        private void FillSettings(ActivityReportModel model)
```

Comment:
You are using same method for all your reports. Seems it can be virtual/override as well

Reply:
Hm, why we are using hardcoded model, when we already have json deserialization?